### PR TITLE
#5 create new trecipe component

### LINKED ISSUE #5: [TC][FE] Implement Add Button
Story Point: 1 

- [ ] Create React component for button

UX mockup:
[create_new_trecipe_button.pdf](https://github.com/shizuko-akamoto/Trecipe/files/4699062/create_new_trecipe_button.pdf)

### PR DESCRIPTION
**Pull request by shizuko-akamoto**
**Changes:**
* Created a Button component with configurable button text and optional fontawesome icon, font size.
Example use of Button component:
For icon names, refer to FontAwesome website, and Step 5 in this link: https://www.digitalocean.com/community/tutorials/how-to-use-font-awesome-5-with-react
`<Button text="Create New" icon="plus-circle" fontSize=16/>`

* Defined color variables for website theme in `src/stylesheets/colors.scss`
* Added some new dependencies

1. Font awesome related packages
2. Lodash: for using js utility methods
3. Enzyme: for use in component unit testing (allows just rendering one component and snapshot testing) https://scotch.io/tutorials/testing-react-components-with-enzyme-and-jest

Closes #5

### CODE REVIEW

File: src/components/button.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * Button props
+ * text: text to display on button
+ * icon: font awesome icon name [optional]
+ * onClick: onClick handler function
+ */
+export interface ButtonProps {
+    text: string;
+    icon: IconProp;
+    onClick(e: MouseEvent<HTMLElement>): void;
```
Comment (shumalll) [inline src/components/button.tsx:16]: Just a question, what determines the argument for onClick? I have string for some cases and don't know if it's appropriate. Should we always use MouseEvent?

Comment (shizuko-akamoto) [reply src/components/button.tsx:16]: That onClick arg is based on onClick attribute on the <button> tag. Just Ctrl+click on "onClick" in `<button className="button" onClick={this.props.onClick} disabled={this.state.disabled}>` you'll see it needs a MouseEvent handler.

File: src/stylesheets/buttons.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@import url('https://fonts.googleapis.com/css2?family=Lato&display=swap');
+
+.button {
+  outline-style: none;
+  background-color: $tc-white;
+  border: 1px solid $silver-gray;
+  box-sizing: border-box;
+  border-radius: 8px;
+  font-style: normal;
+  font-size: 14px;
```
Comment (knox153) [inline src/stylesheets/buttons.scss:11]: Are we making the button fixed size? If we use rem/em, it can scale based on the user's browser font size.

Comment (shizuko-akamoto) [reply src/stylesheets/buttons.scss:11]: I think I'm only fixing the height of the button container? The width should scale itself by the text content. But using em for height is probably good idea 👍

**Comment by shizuko-akamoto** [comment]
> LGTM, but it will be good if we can make the font scalable.

No problemo, will do.

File: src/stylesheets/buttons.scss (hunk 2 above)
Comment (shizuko-akamoto) [reply src/stylesheets/buttons.scss:11]: Made the height: 40px to height: 2.5em.

**Pull request merged**